src/olympia/reviewers/tasks.py

```diff
@@ -22,3 +24,24 @@ def recalculate_post_review_weight(ids):
         for summary in summaries:
             summary.calculate_weight()
             summary.save()
+
+
+def get_recalc_needed_filters():
+    cv = '_current_version__'
+    summary_modified = F(f'{cv}autoapprovalsummary__modified')
+
+    return [
+        # Only recalculate add-ons that received recent abuse reports
+        # possibly through their authors.
+        Q(**{
+            'abuse_reports__created__gte': summary_modified}) |
+        Q(**{
+            'authors__abuse_reports__created__gte': summary_modified
```

Comment:
Because you're going through the relation "manually", you will get deleted abuse reports, which may not be what we want ? It won't do much harm though.

Interestingly, if we get a flood of abuse reports for an add-on, and the add-on gets auto-approved, and they get deleted by admins later, the add-on wouldn't be re-scored. I don't think it's a big issue though, because reviewers will not see the deleted abuse reports so will be able to understand that the situation changed, but maybe worth telling them about this scenario.

Reply:
I excluded deleted reports in 6a6852231c